### ISSUE TITLE
508 Discovery: can Accordion components be passed the appropriate heading level (accessibility)

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#10136]
**NOTE: Brooks offered help in a comment**

## Description

As a Veteran using a screen reader, I want accordion component headers to fit the page heading hierarchy so that I can stay oriented.

Issue brought by Platform team in #10126, see @laflannery comments.

Current state: Accordions seem to always take H2 as their heading level, and this is frequently bad for semantic hierarchy (they display fine):
- On https://www.va.gov/health-care/covid-19-vaccine/ the accordions are H2s under a heading that's H3; should be H4.
- On https://www.va.gov/health-care/eligibility/ the accordions are H2 but should be H3.

## Tasks
- investigate design system Accordion component
- check out a representative variety of Benefits Detail and Resources & Support Detail pages
  - get Wes to help with this audit
- document ideas for template to calculate appropriate level for accordion heading

## Acceptance Criteria
- [x] Review design system docs to see if heading level can be passed in
- [x] Check out code to confirm what's in docs
- [ ] Document path forward in this ticket for two accordion locations: 
  - [ ] Benefits Detail 
  - [ ] Resources & Support Detail
- [ ] Share solution with @laflannery 
- [ ] If component needs changes, ask Platform Design System team whether we can contribute a fix to the DS (or send Dave to bat for this approach)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by bkjohnson** [comment]
Accordion items default to `<h2>` and can be changed by [choosing the heading level and adding a `slot="headline"` attribute](https://github.com/department-of-veterans-affairs/content-build/blob/33d97c363c2d9f3d7296a4ad9a52431cded852ca/src/site/facilities/facility_health_service.drupal.liquid#L11-L13).

**Comment by jtmst** [comment]
Per the design system docs,  accordion items can be passed in a header level with the "level" attribute.  https://design.va.gov/storybook/?path=/docs/components-va-accordion--default

<img width="1043" alt="Screen Shot 2022-08-24 at 11 10 44 AM" src="https://user-images.githubusercontent.com/61624970/186683517-1ecae49d-efa3-4b04-89cc-dd8f8b49decc.png">

This is demonstrated several places in code, one being the Covid vaccine sign up widget.  In the screenshot and code below you can observe that the accordion item is an H4:


<img width="876" alt="Screen Shot 2022-08-24 at 11 09 49 AM" src="https://user-images.githubusercontent.com/61624970/186683895-d0b2b4e6-b145-4a9c-84d2-fcaaffac5137.png">
<img width="780" alt="Screen Shot 2022-08-24 at 11 10 11 AM" src="https://user-images.githubusercontent.com/61624970/186683945-3e0f7e8b-0ee1-43be-b0aa-d4622309d2ab.png">

In vets-website, there are only 20-25 files using the va-accordion component, and a similar number in content-build, which would make auditing header level and adjusting manually a manageable task.

**Comment by bkjohnson** [comment]
@jtmst please don't use the `header` and `level` props and instead use the method I pointed out in my first comment. Those props were added very early and we have since changed direction and they exist only for compatibility with older code. The documentation around this hasn't been a high enough priority for us to work on yet.

**Comment by jilladams** [comment]
@jtmst makes sense with Brooks' note in mind. Could you cut the ticket re: implementing these changes, with whatever details would be needed, and points estimate? I think it makes sense to have you just knock it out in next sprint (or start now if you end up looking for things to do), so ticket doesn't have to be crazy detailed. Can shoot me the link when it's up, and I'll add any necessary add'l tags / epic, etc.
